File: src/my_project_typescript/easy_problems_typescript/from101to150/maximumDifferenceBetweenIncreasingElements.ts
function maximumDifference(nums: number[]): number {
    let answer: number = -1
    let minVal:number = Infinity
    let lenNumber: number = nums.length

    nums.forEach( num => {
        if (num < minVal){
            minVal = num
        }
        else if (num - minVal > 0){
            answer = Math.max(answer, num - minVal)
        }
    })

    return answer 
};

console.log(maximumDifference([7,1,5,4]))

console.log(maximumDifference([9,4,3,2]))

console.log(maximumDifference([1,5,2,10]))

File: src/my_project/easy_problems/from201to250/maximum_difference_between_increasing_elements.py
from typing import List 
import math
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        answer: int = -1
        min_val =  math.inf
        len_nums: int = len(nums)
        
        for i in range(len_nums):
            if nums[i] < min_val:
                min_val = nums[i]
            elif nums[i] - min_val > 0:
                answer = max(answer, nums[i] - min_val)
        
        
        return answer 
    
solution = Solution()

print(solution.maximumDifference(nums = [7,1,5,4]))

print(solution.maximumDifference(nums = [9,4,3,2]))

print(solution.maximumDifference(nums = [1,5,2,10]))
